homeassistant/components/landisgyr_heat_meter/sensor.py

```diff
@@ -56,38 +76,52 @@ def __init__(self, coordinator, description, device):
         self.entity_description = description
 
         self._attr_device_info = device
-        self._attr_should_poll = bool(self.key in ("heat_usage", "heat_previous_year"))
-
-    async def async_added_to_hass(self) -> None:
-        """Call when entity about to be added to hass."""
-        await super().async_added_to_hass()
-        state = await self.async_get_last_sensor_data()
-        if state:
-            self._attr_native_value = state.native_value
+        self.update_values_from_api()
+
+    def update_values_from_api(self):
+        """Update attribute values with the data from the api coordinator."""
+        if self.coordinator.data is None:
+            _LOGGER.debug(
+                "Cannot update %s: could not read any data from the Heat Meter",
+                self.key,
+            )
+            return
```

Comment:
With the suggested `async_config_entry_first_refresh` in the config entry setup, this can't happen.

```suggestion
```

Reply:
Done